app/src/main/java/org/mozilla/tv/firefox/fxa/FxaRepo.kt

```diff
@@ -0,0 +1,123 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.tv.firefox.fxa
+
+import android.content.Context
+import android.util.Log
+import androidx.lifecycle.ProcessLifecycleOwner
+import io.reactivex.subjects.BehaviorSubject
+import kotlinx.coroutines.Deferred
+import mozilla.appservices.fxaclient.Config
+import mozilla.components.concept.sync.AccountObserver
+import mozilla.components.concept.sync.DeviceCapability
+import mozilla.components.concept.sync.DeviceEvent
+import mozilla.components.concept.sync.DeviceEventsObserver
+import mozilla.components.concept.sync.DeviceType
+import mozilla.components.concept.sync.OAuthAccount
+import mozilla.components.concept.sync.Profile
+import mozilla.components.service.fxa.manager.DeviceTuple
+import mozilla.components.service.fxa.manager.FxaAccountManager
+import org.mozilla.tv.firefox.fxa.FxaRepo.AccountState.AUTHENTICATED
+import org.mozilla.tv.firefox.fxa.FxaRepo.AccountState.AUTHENTICATED_NO_PROFILE
+import org.mozilla.tv.firefox.fxa.FxaRepo.AccountState.NEEDS_REAUTHENTICATION
+import org.mozilla.tv.firefox.fxa.FxaRepo.AccountState.NOT_AUTHENTICATED
+
+private const val LOGTAG = "FxaRepo"
+
+/**
+ * Manages Firefox Account (FxA) state. In particular, [accountStateDontUseMeYet] exposes the current sign in
+ * state of the user. If you want to initiate sign in, see [FxaLoginUseCase.beginLogin].
+ *
+ * Devs should use this class rather than interacting with the FxA library directly.
```

Comment:
I know we haven't done this in the past, but it's less work than it sounds, and might be a good option here.

If we don't want to expose the FxA library to the rest of the app, maybe we should move this code into a separate Gradle module.  We can choose not to expose the dependency transitively, and instead only provide access to our repo.  

The setup/complexity for this is pretty minimal, it enforces some separation, and if we made it a habit it would even improve build times (not that those are a major concern for us yet).  I've been playing with this in a side app and I'm liking it quite a bit.

Reply:
I think about this from time-to-time for our other integrations (Sentry, Picasso, Telemetry, etc.). However, I'm not sure the benefits:
- (potentially?) faster build times
- Clean separation of who should use what dependencies
- (I think there are other small benefits here too but I'm pressed for time)

Outweigh the downsides:
- Have to rewrite all data types from the library (e.g. can't easily expose just FxaAccountManager if we wanted to without exposing entire dependency)
- afaik, IDE does not make it clear why objects can't be imported (both app/ to thirdparty dependencies and our interface to app/) and newer devs may stumble on this
- Similar files (e.g. FxaRepo & SessionRepo) are in entirely separate file hierarchies and can be hard to browse in IDE or file system if you forget to take module boundaries into account
- More complex build code: it must be multi-module aware (and we struggle with build code already)
- Helper classes (such as utils) are not yet broken into a "common" gradle module so they're unavailable to our intermediate build modules/interfaces.

@Baron-Severin Briefly, what do you think? We should talk IRL for a longer conversation.

Reply:
Sounds worth talking about in person.  I could go either way on it.